ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetBlockImporter.java

```diff
@@ -51,41 +43,23 @@ public synchronized boolean importBlock(
       final Block block,
       final HeaderValidationMode headerValidationMode,
       final HeaderValidationMode ommerValidationMode) {
-    final BlockHeader header = block.getHeader();
-
-    final Optional<BlockHeader> maybeParentHeader =
-        context.getBlockchain().getBlockHeader(header.getParentHash());
-    if (!maybeParentHeader.isPresent()) {
-      LOG.error(
-          "Attempted to import block {} with hash {} but parent block {} was not present",
-          header.getNumber(),
-          header.getHash(),
-          header.getParentHash());
-      return false;
-    }
-    final BlockHeader parentHeader = maybeParentHeader.get();
 
-    if (!blockHeaderValidator.validateHeader(header, parentHeader, context, headerValidationMode)) {
-      return false;
-    }
+    final Optional<BlockValidationOutputs> outputs =
+        blockValidator.validateAndProcessBlock(
+            context, block, headerValidationMode, ommerValidationMode);
 
-    final MutableBlockchain blockchain = context.getBlockchain();
-    final MutableWorldState worldState =
-        context.getWorldStateArchive().getMutable(parentHeader.getStateRoot());
-    final BlockProcessor.Result result = blockProcessor.processBlock(blockchain, worldState, block);
-    if (!result.isSuccessful()) {
-      return false;
-    }
+    outputs.ifPresent(o -> persistState(o, block, context));
```

Comment:
It would be great to avoid using single letter variable names. Always takes me ages to work out what the heck `o` actually is.

Reply:
done.